server/src/main/java/com/skillmagnet/User/UserController.java

```diff
@@ -5,22 +5,45 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.time.LocalDateTime;
+
 @RestController
 @RequestMapping("api")
 public class UserController  {
 
     @Autowired
     UserRepository userRepository;
 
-    @PostMapping("/user")
-    public ResponseEntity<User> createUser(@RequestParam("username") String username) {
-        User newUser = new User(username);
+    /**
+     * Creates a new user out of the given username and password.
+     * @param username - string containing username
+     * @param password - string containing plaintext password
+     * @return
+     *          Success - The Created User
+     *          Failure - BAD_REQUEST (Username Taken)
+     */
+    @PostMapping("/user/create")
+    public ResponseEntity<User> createUser(@RequestParam("username") String username,
+                                           @RequestParam("password") String password) {
+        // Username taken check
+        if(userRepository.findByUsername(username) == null) {
+            User newUser = new User(username, password);
+
+            userRepository.save(newUser);
 
-        userRepository.save(newUser);
+            return new ResponseEntity<>(newUser, HttpStatus.BAD_REQUEST);
+        }
 
-        return new ResponseEntity<>(newUser, HttpStatus.OK);
+        return new ResponseEntity<>(null, HttpStatus.OK);
```

Comment:
:wrench: I think you accidentally swapped your response statuses. Successful create user should be `HttpStatus.OK`.

Also can we make the duplicate username its own seperate check. As in validate the username doesn't exist. If it does then return a 400 with error message that says something like 'Duplicate username'. Then try to create user  and if that fails then u can just return the null response.

Reply:
For the response when user creation fails (null response body), what should the HTTPStatus be, rn i have as OK

Reply:
500 i.e. internal server error. If it fails at that point it's our fault, not bad data that the client passed in